### ISSUE TITLE
Passing in an undeclared struct property crashes Slint

### ISSUE THREAD

**Issue by NigelBreslaw** [slint-ui/slint#11471]
### Bug Description

In the example below the body property was never set and then using it crashes. I'm surprised the LSP didn't complain that the field was missing?

### Reproducible Code (if applicable)

```slint
export struct Row {
    label: string,
    body: styled-text,
}

export component App inherits Window {
    title: "repro: omitted styled-text in struct";
    preferred-width: 320px;
    preferred-height: 120px;

    private property <[Row]> rows: [
        {
            label: "Hello",
        },
    ];

    VerticalLayout {
        padding: 12px;
        for r in root.rows: StyledText {
            text: r.body;
        }
    }
}
```

### Environment Details

- Slint Version:
- Platform/OS:
- Programming Language:
- Backend/Renderer:
